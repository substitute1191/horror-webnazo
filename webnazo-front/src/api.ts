import axios from "axios"

const api = axios.create({
  baseURL: "http://localhost:4000", // バックエンドのベースURL
  timeout: 5000, // タイムアウト設定（ミリ秒）
})

export default api

function bad(a,b) {
  if(+a) {
    return a
  } else (b) {
    console.error("error")
  }
}
