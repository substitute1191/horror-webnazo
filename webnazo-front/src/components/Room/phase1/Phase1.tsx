/* eslint-disable max-lines-per-function */
import rankmatchLogo from "@/assets/image/rankmatch_logo.png"

import useProceed from "../useProceed"
import useBGM from "@/SoundManager/useBGM"
import bgmSrc from "@/assets/sound/pom_pom_shower.mp3"
import { useEffect } from "react"
import SiteDescription from "../components/prehorror/SiteDescription"
import HighAchierver from "../components/prehorror/HighAchiever"
import AboutPyramid from "../components/prehorror/AboutPyramid"
import Precautions from "../components/prehorror/Precautions"

const Phase1 = () => {
  const { proceed } = useProceed()
  const { play, pause } = useBGM(bgmSrc)

  useEffect(() => {
    void play()
    return () => {
      pause()
    }
  })

  return (
    <div className="bg-yumekawa bg-white/40 bg-cover bg-blend-color">
      <div className="font-pop mx-auto w-full border-2 border-solid border-fuchsia-200 bg-gradient-to-t from-orange-200 via-lime-300 to-emerald-200 pb-52 pt-7 lg:w-3/5">
        <h1>
          <img
            src={rankmatchLogo}
            alt="ランクマッチのロゴ"
            className="animate-float mx-auto mb-5 w-1/2"
          />
        </h1>
        <div className="mx-auto w-11/12 rounded-2xl border border-solid border-fuchsia-200 bg-white px-2 py-2">
          <div className="rounded-2xl border-2 border-dashed border-fuchsia-200 px-12 py-5">
            <SiteDescription />
            <div className="mb-5 mt-14 text-center text-3xl font-extrabold">
              <div>
                次回のゲームは<span className="text-red-500">8月1日21時</span>
                からスタート！
              </div>
              今回は
              <span className="underline-offset-3 text-red-500 underline">
                2人1組
              </span>
              での参加となります！友人を誘って参加しよう！
            </div>
            <div className="text-center">
              <button
                className="flex w-full transform items-center justify-between rounded-full bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500 px-3 py-7 text-center text-7xl font-bold text-white shadow-lg transition duration-300 hover:scale-105 hover:shadow-2xl focus:outline-none"
                onClick={() => proceed(2)}
              >
                <span className="flex-grow text-center">ゲームに参加する</span>
                <i className="fas fa-angle-right mr-5"></i>
              </button>
            </div>
            <HighAchierver />
            <AboutPyramid />
            <Precautions />
          </div>
        </div>
      </div>
    </div>
  )
}
export default Phase1
