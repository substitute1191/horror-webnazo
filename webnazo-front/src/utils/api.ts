import axios from "axios"

const api = axios.create({
  baseURL: "http://localhost:4000/api", // バックエンドのベースURL
  timeout: 5000, // タイムアウト設定（ミリ秒）
  /* eslint-disable @typescript-eslint/naming-convention */
  headers: {
    "Content-Type": "application/json",
  },
})

export default api
