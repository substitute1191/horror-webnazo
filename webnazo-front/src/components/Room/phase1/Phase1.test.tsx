import Phase1 from "./Phase1"
import { render, screen, waitFor } from "@testing-library/react"
import api from "@/utils/api"
import userEvent from "@testing-library/user-event"
import { act } from "react"

jest.mock("@/assets/image/rankmatch_logo.png", () => "rankmatch_logo.png")
jest.mock("@/assets/image/mascot/mascot.png", () => "mascot.png")
jest.mock("@/assets/image/ranking/rank_1.png", () => "rank_1.png")
jest.mock("@/assets/image/ranking/rank_2.png", () => "rank_2.png")
jest.mock("@/assets/image/ranking/rank_3.png", () => "rank_3.png")

jest.mock("@/assets/sound/pom_pom_shower.mp3", () => "mocked-bgm-file")
jest.mock("@/assets/sound/決定ボタンを押す53.mp3", () => "mocked-bgm-file")

const mockPlay = jest.fn(() => Promise.resolve())
const mockPause = jest.fn()
jest.mock("@/SoundManager/useBGM", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    play: mockPlay,
    pause: mockPause,
  })),
}))

jest.mock("@/SoundManager/useSE", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    play: mockPlay,
    pause: mockPause,
  })),
}))

const mockRoomId = "test-room-id"
jest.mock("react-router-dom", () => ({
  useParams: jest.fn(() => ({ roomId: mockRoomId })),
}))

describe("Phase1 component", () => {
  const user = userEvent.setup()
  const apiPost = jest.spyOn(api, "post")

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it("マウント時とアンマウント時にそれぞれplay()とpause()が実行されることの確認", () => {
    const { unmount: u } = render(<Phase1 />)
    const unmount = u
    expect(mockPlay).toHaveBeenCalledTimes(1)

    act(() => {
      unmount()
    })
    expect(mockPause).toHaveBeenCalledTimes(1)
  })

  it("子コンポーネントのレンダリング確認", () => {
    render(<Phase1 />)
    const siteDescription = screen.getByText("このサイトは？")
    const highAchiever = screen.getByText("知的生命体YOSHIO")
    const aboutPyramid = screen.getByText("ピラミッド君")
    const precautions = screen.getByText("注意事項")

    expect(siteDescription).toBeInTheDocument()
    expect(highAchiever).toBeInTheDocument()
    expect(aboutPyramid).toBeInTheDocument()
    expect(precautions).toBeInTheDocument()
  })

  it("「ゲームに参加する」ボタンを押すとproceedが呼ばれる", async () => {
    ;(api.post as jest.MockedFunction<typeof api.post>).mockResolvedValue({
      data: { phase: 2 },
    })

    render(<Phase1 />)

    const joinGame = screen.getByText("ゲームに参加する")
    expect(joinGame).toBeInTheDocument()

    await user.click(joinGame)

    await waitFor(() => {
      expect(apiPost).toHaveBeenCalledWith(`/room/${mockRoomId}/proceed`, {
        phase: 2,
      })
    })
  })

  it("コンポーネントのスナップショットが一致する", () => {
    const { container } = render(<Phase1 />)
    expect(container).toMatchSnapshot()
  })
})
