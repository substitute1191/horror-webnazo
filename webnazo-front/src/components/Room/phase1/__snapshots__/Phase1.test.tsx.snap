// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phase1 component コンポーネントのスナップショットが一致する 1`] = `
<div>
  <div
    class="bg-yumekawa bg-white/40 bg-cover bg-blend-color"
  >
    <div
      class="font-pop mx-auto w-full border-2 border-solid border-fuchsia-200 bg-gradient-to-t from-orange-200 via-lime-300 to-emerald-200 pb-52 pt-7 lg:w-3/5"
    >
      <h1>
        <img
          alt="ランクマッチのロゴ"
          class="animate-float mx-auto mb-5 w-1/2"
          src="rankmatch_logo.png"
        />
      </h1>
      <div
        class="mx-auto w-11/12 rounded-2xl border border-solid border-fuchsia-200 bg-white px-2 py-2"
      >
        <div
          class="rounded-2xl border-2 border-dashed border-fuchsia-200 px-12 py-5"
        >
          <h2
            class="pb-5 text-center text-6xl tracking-widest"
          >
            このサイトは？
          </h2>
          <p
            class="bg-teal-100 p-8 text-left text-2xl"
          >
            このサイトでは毎週所定の時間に「
            <span
              class="text-red-500"
            >
              ナゾトキランクマッチ
            </span>
            」が開催される！
            <br />
            皆で一斉に同じ謎解き問題に参加して、解くまでの時間を競うのだ！解くのが早ければ早いほどレーティングが上がっていくぞ！
            <br />
            <span
              class="text-red-500"
            >
              全国の謎解きマニアたちと勝負せよ！
            </span>
          </p>
          <div
            class="mb-5 mt-14 text-center text-3xl font-extrabold"
          >
            <div>
              次回のゲームは
              <span
                class="text-red-500"
              >
                8月1日21時
              </span>
              からスタート！
            </div>
            今回は
            <span
              class="underline-offset-3 text-red-500 underline"
            >
              2人1組
            </span>
            での参加となります！友人を誘って参加しよう！
          </div>
          <div
            class="text-center"
          >
            <button
              class="flex w-full transform items-center justify-between rounded-full bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500 px-3 py-7 text-center text-7xl font-bold text-white shadow-lg transition duration-300 hover:scale-105 hover:shadow-2xl focus:outline-none"
            >
              <span
                class="flex-grow text-center"
              >
                ゲームに参加する
              </span>
              <i
                class="fas fa-angle-right mr-5"
              />
            </button>
          </div>
          <div
            class="mb-20 mt-16 bg-pink-100 p-12"
          >
            <h3
              class="mb-7 text-6xl"
            >
              前回の成績優秀者
            </h3>
            <ul
              class="leading-loose"
            >
              <li
                class="flex text-6xl"
              >
                <img
                  alt="一位"
                  class="w-48"
                  src="rank_1.png"
                />
                <span
                  class="mb-10 mt-auto"
                >
                  知的生命体YOSHIO 
                  <span
                    class="text-4xl"
                  >
                    さん
                  </span>
                </span>
              </li>
              <li
                class="mb-7 flex pl-8 text-5xl"
              >
                <img
                  alt="二位"
                  class="w-32"
                  src="rank_2.png"
                />
                <span
                  class="mb-7 ml-3 mt-auto"
                >
                  北亨夏高校クイ研 
                  <span
                    class="text-[2rem]"
                  >
                    さん
                  </span>
                </span>
              </li>
              <li
                class="flex pl-14 text-4xl"
              >
                <img
                  alt="三位"
                  class="w-20"
                  src="rank_3.png"
                />
                <span
                  class="mb-4 ml-7 mt-auto"
                >
                  ゆ（＠ω＠） 
                  <span
                    class="text-3xl"
                  >
                    さん
                  </span>
                </span>
              </li>
            </ul>
          </div>
          <div
            class="mb-20 bg-red-50 p-12"
          >
            <img
              alt="ピラミッド君"
              class="mx-auto mb-10"
              src="mascot.png"
            />
            <h3
              class="text-center text-5xl"
            >
              ピラミッド君
            </h3>
            <p
              class="mt-8 text-2xl"
            >
              このサイトのマスコットキャラクター。謎で頂点を目指すユーザーの皆を応援している。可愛い外見で毒のあることを言うという自己認識を持っているが、そもそも自分が可愛くないことに気づいていない。
              <br />
              オカルトが大の苦手であり、心霊スポットなどお化けが出そうなところからはすぐに逃げ出してしまう。
              <br />
              そのためピラミッドが王族の墓として作られたという通説を真っ赤なデマであるとして強く否定している。
            </p>
          </div>
          <div
            class="bg-orange-100 p-12"
          >
            <h3
              class="mb-10 text-center text-5xl"
            >
              注意事項
            </h3>
            <ul
              class="text-2xl"
            >
              <li
                class="list-inside list-disc"
              >
                シーズン中は謎の回答をSNSなどでネタバレしないようにしてください。
              </li>
              <br />
              <li
                class="list-inside list-disc"
              >
                このサイトは同時に多くのユーザーが閲覧しています。過度な更新を行うとサーバーに過剰な負荷がかかる恐れがあるため、ページが表示されない時にたくさん更新はしないでください。
              </li>
              <br />
              <li
                class="list-inside list-disc"
              >
                サイトが落ちてしまったときは、5分程度で正常に戻りますので更新せずにそのままお待ちください。
              </li>
              <br />
              <li
                class="list-inside list-disc"
              >
                ソースコードを閲覧して解く行為はお控えください。
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
